### PR TITLE
Remove `JvmLocalCache` now that everything is round-trippable on its own

### PR DESCRIPTION
**Pull request by nedtwigg**
- Capstone piece of #1274

### CODE REVIEW

File: plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTask.java [hunk 1]
```diff
@@ -155,22 +150,17 @@ public File getOutputDirectory() {
...
 	}
 
 	public void setSteps(List<FormatterStep> steps) {
-		this.steps.set(PluginGradlePreconditions.requireElementsNonNull(steps));
-	}
-
-	public boolean addStep(FormatterStep step) {
-		return this.steps.get().add(Objects.requireNonNull(step));
+		this.steps.clear();
+		PluginGradlePreconditions.requireElementsNonNull(steps);
```
Comment (Goooler) [inline plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTask.java:162]:
```suggestion
    PluginGradlePreconditions.requireElementsNonNull(steps);
		this.steps.clear();
```

**Review by Goooler — APPROVED** [review]

File: plugin-gradle/src/test/java/com/diffplug/gradle/spotless/DiffMessageFormatterTest.java [hunk 2]
```diff
@@ -151,10 +151,10 @@ void customRunToFixMessage() throws Exception {
 	@Test
 	void whitespaceProblem() throws Exception {
 		Bundle spotless = create(setFile("testFile").toContent("A \nB\t\nC  \n"));
-		spotless.task.addStep(FormatterStep.createNeverUpToDate("trimTrailing", input -> {
+		spotless.task.setSteps(List.of(FormatterStep.createNeverUpToDate("trimTrailing", input -> {
```
Comment (jbduncan) [inline plugin-gradle/src/test/java/com/diffplug/gradle/spotless/DiffMessageFormatterTest.java:154]: I'm tempted to revert this code change and all other `addStep` -> `setStep` changes. WDYT?

**Review by jbduncan — COMMENTED** [review]
Other than @Goooler's comment and my nit-picking comment below, this PR LGTM!

File: plugin-gradle/src/test/java/com/diffplug/gradle/spotless/DiffMessageFormatterTest.java (hunk 2 above)
Comment (nedtwigg) [reply plugin-gradle/src/test/java/com/diffplug/gradle/spotless/DiffMessageFormatterTest.java:154]: I can see an argument for `setSteps(FormatterStep... steps)` to remove the manual `List.of`, but I think the mutable-naming `addStep` is bad. It's used only for a few tests, which is why I just did the `List.of`.

Comment (jbduncan) [reply plugin-gradle/src/test/java/com/diffplug/gradle/spotless/DiffMessageFormatterTest.java:154]: Fair enough! I have no strong opinions either way, so I'm happy to continue with `setSteps`.

**Pull request merged**